data/software_stack/eessi.yaml

```diff
@@ -23,5 +23,6 @@ profile::cvmfs::client::lmod_default_modules:
 
 profile::squid::server::cvmfs_acl_regex:
   - '^(.*\.cvmfs\.eessi-infra\.org)$'
+  - '^(.*s1\.eessi\.science)$'
 
-profile::gpu::install::lib_symlink_path: '/opt/eessi/lib'
+profile::gpu::install::lib_symlink_path: '/opt/eessi/nvidia/x86_64/host'
```

Comment:
Ideally I'd use `$(uname -m)` to give me the arch family here

Reply:
We can use puppet fact in hieradata, i.e: `%{os.architecture}`

Reply:
Even if we do this, it's still not enough for EESSI, we implemented GPU support for the stack and it basically requires that someone runs the script https://github.com/EESSI/software-layer/blob/2023.06-software.eessi.io/scripts/gpu_support/nvidia/link_nvidia_host_libraries.sh which does the symlinking, but allows for the potential use of CUDA compatibility libraries, and also places (a symlink to) the libraries in a trusted location for the Gentoo Prefix linker. 

I'm not sure what the best approach here is, for the mentioned script to work, you already need a working CVMFS install. Could there be support for a post install script for CVMFS?

Reply:
In the next EESSI release we would plan to add additional trusted locations for the linker (right now there is only one):

- A general one like we have currently
- One each for NVIDIA GPUs and AMD GPUs (and perhaps even Intel GPUs)

Reply:
For now, I think I'll remove the symlink line so that this PR can be merged, and I'll fork the GPU support of EESSI into an issue.

Reply:
Moved this discussion to https://github.com/ComputeCanada/puppet-magic_castle/issues/301